lib/dash/dash_parser.js

```diff
@@ -528,25 +588,240 @@ shaka.dash.DashParser = class {
       // maintain consistency from here on.
       presentationTimeline.lockStartTime();
     } else {
-      // Just update the variants and text streams, which may change as periods
-      // are added or removed.
-      this.manifest_.variants = this.periodCombiner_.getVariants();
-      const textStreams = this.periodCombiner_.getTextStreams();
-      if (textStreams.length > 0) {
-        this.manifest_.textStreams = textStreams;
+      await this.postPeriodProcessing_(periodsAndDuration.periods, false);
+    }
+  }
+
+  /**
+   * Handles common procedures after processing new periods.
+   *
+   * @param {!Array<shaka.extern.Period>} periods to be appended
+   * @param {boolean} isPatchUpdate does call comes from mpd patch update
+   * @private
+   */
+  async postPeriodProcessing_(periods, isPatchUpdate) {
+    await this.periodCombiner_.combinePeriods(periods, true, isPatchUpdate);
```

Comment:
This is another call to combinePeriods that wasn't in the original flow, executed even when `enablePatchMPDSupport` is false or `patchLocationUrl_` is `null`.

Reply:
This is the only call now during an update.